qiskit_rigetti/_qcs_job.py

```diff
@@ -107,6 +109,28 @@ def _start_circuit(self, circuit: QuantumCircuit) -> Response:
         # typing: QuantumComputer's inner QAM is generic, so we set the expected type here
         return cast(Response, self._qc.qam.execute(executable))
 
+    def _handle_barriers(self, qasm: str) -> str:
+        lines = []
+        for line in qasm.splitlines():
+            if not line.startswith("barrier"):
+                lines.append(line)
+                continue
+
+            qubits = line[(len("barrier ")) :]
+            num_qubits = qubits.count(",") + 1
```

Comment:
This might not be the full picture. A barrier could look like `barrier q;` with `qreg q[2];`. In the above, that would count for just a single qubit where `q` is actually two.

Reply:
I thought of that too when looking at the QASM spec. However, I wasn't able to produce this effect when going through qiskit circuits, as adding `circuit.barrier(qreg)` still caused the individual qubits to be listed. What that means is that I could cover this case, but there'd be no real way to test it... at least not in a way that reflects how a user would come across this scenario.

Reply:
What I'll do is at least add a test for the `circuit.barrier(qreg)` in case qiskit's implementation ever changes and results in the qasm you mentioned

Reply:
Test added @notmgsk

Reply:
Ah! Well, I keep forgetting that we're integrating with _qiskit_ rather than _qasm_, and so it makes sense that if qiskit expands these registers (I *think* that's what you're saying) then it's not a problem. (I should've known you had already considered it :D )